diff -ru a/vncviewer/DesktopWindow.cxx b/vncviewer/DesktopWindow.cxx
--- a/vncviewer/DesktopWindow.cxx	2017-05-16 15:53:28.000000000 +0200
+++ b/vncviewer/DesktopWindow.cxx	2017-07-04 15:08:29.448166409 +0200
@@ -658,14 +658,6 @@
 
 void DesktopWindow::menuOverlay(void* data)
 {
-  DesktopWindow *self;
-
-  self = (DesktopWindow*)data;
-
-  if (strcmp((const char*)menuKey, "") != 0) {
-    self->setOverlay(_("Press %s to open the context menu"),
-                     (const char*)menuKey);
-  }
 }
 
 void DesktopWindow::setOverlay(const char* text, ...)
